core/common/src/main/java/alluxio/worker/WorkerInquireClient.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker;
+
+import alluxio.exception.status.UnavailableException;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Client for determining a worker is serving RPCs.
+ */
+@ThreadSafe
+public interface WorkerInquireClient {
```

Comment:
We have `MasterInquireClient` because sometimes (e.g. in HA mode with zookeeper), it can be non-trivial to determine the active RPC address. It's easier for workers because the RPC address never changes, so we don't need a client for inquiring what the RPC address should be.

This is more like a health check where we just want to see whether the worker is serving RPCs. Perhaps rename to WorkerHealthCheck and support a `boolean isServing()` method?

Reply:
Yes, make more sense in that way, done.